### DISCUSSION TITLE
Need Discrete Input Help

### DISCUSSION THREAD

**Discussion by DrAnonymous** [forkineye/ESPixelStick#450] [Q&A]
Is there an I/O pin that can be used for a button input? I'm assuming this would require soldering to one of the 8 pin headers. I want to create a self contained display that has a single button to cycle modes.

Looking here: 
[Wemos Pinout](https://i2.wp.com/randomnerdtutorials.com/wp-content/uploads/2019/05/ESP8266-WeMos-D1-Mini-pinout-gpio-pin.png?quality=100&strip=all&ssl=1)

Is ADC0 or GPIO3 available?
Is GPIO2 dedicated to driving pixels (WS2811 in my case)?

Trying to make a gift for my father, so any help is appreciated.

**Comment by DrAnonymous** [comment]
My apologizes for accidentally duplicating the question. Couldn't figure out how to delete the prior one. :(

**Comment by forkineye** [comment]
There is some code for handling an external input - https://github.com/forkineye/ESPixelStick/tree/main/ESPixelStick/src/input but you would have to write the logic for iterating through the modes / playlist.

**Reply by DrAnonymous** [reply]
Reading through ExternalInput.cpp and InputMgr.cpp, it appears there's already logic hooked up to read a button input. It switches between effects for a short press and turns on/off effects for a long press. I feel silly for not seeing that previously.  

What DIO is it using by default?  
I don't see anything in the web pages, even though there is code to pass config and state info via JSON in the src code. Does something need to be done to make it display?

**Reply by MartinMueller2003** [reply]
Shelby asked me to deactivate that code. You can easily turn it back on and choose any IO pin you want.
